sagemaker-python-sdk/pytorch_mnist/mnist.py

```diff
@@ -195,4 +203,18 @@ def save_model(model, model_dir):
     parser.add_argument('--data-dir', type=str, default=os.environ['SM_CHANNEL_TRAINING'])
     parser.add_argument('--num-gpus', type=int, default=os.environ['SM_NUM_GPUS'])
 
-    train(parser.parse_args())
+    args = parser.parse_args()
+    
+    master = json.loads(os.environ['SM_TRAINING_ENV'])['master_hostname']
```

Comment:
why not just use args.hosts[0] since it doesn't really make a difference for the example?
the `master_hostname` also [just chooses the first host in the list](https://github.com/aws/sagemaker-training-toolkit/blob/1ae483914f7b4fc8f9ddf3b98060d70da1b45750/src/sagemaker_training/environment.py#L562) (this variable is used internally for setting up horovod)